lib/circuitry/cli.rb

```diff
@@ -0,0 +1,66 @@
+require 'circuitry/provisioner'
+require 'thor'
+
+module Circuitry
+  class CLI < Thor
+    class_option :verbose, aliases: :v, type: :boolean
+
+    desc 'provision <queue> -t <topic> [<topic> ...]', 'Provision a queue subscribed to one or more topics'
+
+    long_desc <<-END
+      Creates an SQS queue and dead letter queue named <queue>[-failures] with appropriate
+      SNS access policy along with one or more SNS topics named <topic> that has an SQS subscription for
+      each.
+
+      When the queue already exists, its policy will be added or updated.
+
+      When a topic already exists, it will be ignored.
+
+      When a topic subscription already exists, it will be ignored.
+    END
+
+    option :topics, aliases: :t, type: :array, required: :true
+    option :access_key, aliases: :a
+    option :secret_key, aliases: :s
+    option :region, aliases: :r
+
+    def provision(queue_name)
+      with_custom_config(queue_name) do |config|
+        logger = Logger.new(STDOUT)
+        logger.level = Logger::INFO if options['verbose']
+        Circuitry::Provisioner.new(config, logger).run
+      end
+    end
+
+    private
+
+    def say(*args)
+      puts(*args) if options['verbose']
+    end
+
+    def with_custom_config(queue_name, &block)
+      original_access_key = Circuitry.config.access_key
+      original_secret_key = Circuitry.config.access_key
+      original_region = Circuitry.config.region
+      original_subscriber_queue_name = Circuitry.config.subscriber_queue_name
+      original_dead_letter_subscriber_queue_name = Circuitry.config.subscriber_dead_letter_queue_name
+      original_publisher_topic_names = Circuitry.config.publisher_topic_names
+
+      Circuitry.config.access_key = options.fetch('access_key', original_access_key)
+      Circuitry.config.secret_key = options.fetch('secret_key', original_secret_key)
+      Circuitry.config.region = options.fetch('region', original_region)
+      Circuitry.config.subscriber_queue_name = queue_name
+      Circuitry.config.subscriber_dead_letter_queue_name = "#{queue_name}-failures"
```

Comment:
Ahh, hmm...it'd be nice to allow that for consistency.  I don't think it'd be too difficult to add, just another command-line option.  I think I included it in my original PR, and we could still fallback on your default value if not supplied.